engine/execution/ingestion/engine.go

```diff
@@ -996,8 +996,12 @@ func (e *Engine) matchOrRequestCollections(
 			Hex("collection_id", logging.ID(guarantee.ID())).
 			Msg("requesting collection")
 
+		guarantors, err := protocol.FindGuarantors(e.state, guarantee)
```

Comment:
@m4ksio does this function differentiate between byzantine inputs and unexpected internal errors already? If so, we would need proper error treatment here.